app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/PastWorkoutsActivity.java

```diff
@@ -1,29 +1,104 @@
 package edu.drexel.lapcounter.lapcounter.frontend;
 
+import android.arch.lifecycle.ViewModelProviders;
 import android.content.Intent;
 import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
 import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
 import android.view.View;
 
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TimeZone;
+
 import edu.drexel.lapcounter.lapcounter.R;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.WorkoutViewModel;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.Workout;
 import edu.drexel.lapcounter.lapcounter.frontend.navigationbar.NavBar;
 
 public class PastWorkoutsActivity extends AppCompatActivity {
     private final NavBar mNavBar = new NavBar(this, R.id.navigation_past_workouts);
+    private String tempStartDate;
+
+    private WorkoutViewModel mWorkoutViewModel;
+
+
+
+    private static final String TAG = "PastWorkoutsActivity";
+
+    List<Workout> allWorkoutsDesc;
+    private ArrayList<String> mWorkoutDate = new ArrayList<>();
+    private ArrayList<String> mPoolLength = new ArrayList<>();
+    private ArrayList<String> mID = new ArrayList<>();
+
+
+
+
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        mWorkoutViewModel = ViewModelProviders.of(this).get(WorkoutViewModel.class);
+
+
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_past_workouts);
-
+        Log.d(TAG, "onCreate: started.");
         // In the final version, use R.string.<string id> for titles
         getSupportActionBar().setTitle("Past Workouts");
 
+
+        try {
+            allWorkoutsDesc =  mWorkoutViewModel.getAllWorkoutsDecending();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+
+        for (Workout wrkout : allWorkoutsDesc) {
+            initPastWorkoutsView(wrkout);
+        }
+
+
+        initRecyclerView();
         mNavBar.init();
     }
 
-    public void selectWorkout(View view) {
-        Intent intent = new Intent(this, WorkoutDetailsActivity.class);
-        startActivity(intent);
+
+    private void initRecyclerView(){
```

Comment:
Not sure where to put this comment but it deals with the Past Workouts screen.  If we have no data in the app to start with, we do not have the green button that allows them to go start a new workout.